torch/onnx/symbolic_opset9.py

```diff
@@ -2745,3 +2745,13 @@ def as_strided(g, self, sizes, strides, offset=None):
         if offset:
             ind = g.op("Add", ind, g.op("Constant", torch.tensor([offset])))
         return g.op("Gather", self_1d, ind)
+
+
+def __derive_index(g, index, start, step):
+    return g.op("Add", start, g.op("Mul", index, step))
+
+
+def __range_length(g, lo, hi, step):
```

Comment:
Consider adding link to source in comment, and copying the code of computation as well for reference. https://github.com/pytorch/pytorch/blob/671ee71ad4b6f507218d1cad278a8e743780b716/torch/csrc/jit/runtime/register_prim_ops.cpp#L186
```c++
           if (step > 0 && lo < hi) {
             push(stack, 1 + (hi - 1 - lo) / step);
           } else if (step < 0 && lo > hi) {
             push(stack, 1 + (lo - 1 - hi) / (0 - step));
           } else {
             push(stack, 0);
           }
```
The current symbolic should be equivalent with the positive case, is it also equivalent with the case when (hi-lo) and step are both negative?

Reply:
This is covered in tests for: 
```for idx in range(len(x) - 1, -len(x), -2):```